libraries/LittleFS/src/LittleFS.h

```diff
@@ -0,0 +1,446 @@
+        int rc = lfs_file_seek(_fs->getFs(), _getFd(), offset, (int)mode); // NB. SeekMode === LFS_SEEK_TYPES
+        if (rc < 0) {
+            DEBUGV("lfs_file_seek rc=%d\n", rc);
+            return false;
+        }
+        return true;
+    }
+
+    size_t position() const override {
+        if (!_opened) {
+            return 0;
+        }
+        int result = lfs_file_tell(_fs->getFs(), _getFd());
+        if (result < 0) {
+            DEBUGV("lfs_file_tell rc=%d\n", result);
+            return 0;
+        }
+
+        return result;
+    }
+
+    size_t size() const override {
+        return _opened? lfs_file_size(_fs->getFs(), _getFd()) : 0;
+    }
+
+    void close() override {
+        if (_opened) {
+            lfs_file_close(_fs->getFs(), _getFd());
+            _opened = false;
+            DEBUGV("lfs_file_close: fd=%p\n", _getFd());
+        }
+    }
+
+    const char* name() const override {
+        return _opened ? (const char*) _name : nullptr;
+    }
+
+protected:
+    lfs_file_t *_getFd() const {
+        return _fd.get();
+    }
+
+    LittleFSImpl                *_fs;
+    std::shared_ptr<lfs_file_t>  _fd;
+    char                         _name[LFS_NAME_MAX+1];
+    bool                         _opened;
+};
+
+class LittleFSDirImpl : public DirImpl
+{
+public:
+    LittleFSDirImpl(const String& pattern, LittleFSImpl* fs, std::shared_ptr<lfs_dir_t> dir)
+        : _pattern(pattern) , _fs(fs) , _dir(dir) , _valid(false), _opened(true) {
+        memset(&_dirent, 0, sizeof(_dirent));
+    }
+
+    ~LittleFSDirImpl() override {
+        if (_opened) {
+            lfs_dir_close(_fs->getFs(), _getDir());
+        }
+    }
+
+    FileImplPtr openFile(OpenMode openMode, AccessMode accessMode) override {
+        if (!_valid) {
+            return FileImplPtr();
+        }
+        return _fs->open(fileName(), openMode, accessMode);
+    }
+
+    const char* fileName() override {
+        if (!_valid) {
+            return nullptr;
+        }
+        return (const char*) _dirent.name;
+    }
+
+    size_t fileSize() override {
+        if (!_valid) {
+            return 0;
+        }
+        return _dirent.size;
+    }
+
+    bool next() override {
+        const int n = _pattern.length();
+        bool match;
+        do {
+            _dirent.name[0] = 0;
+            int rc = lfs_dir_read(_fs->getFs(), _getDir(), &_dirent);
+            _valid = (rc == 1);
+            LittleFSImpl::lfsNameToSPIFFS(_dirent.name, _dirent.name);
+            match = (!n || !strncmp((const char*) _dirent.name, _pattern.c_str(), n));
+        } while (_valid && !match);
+        return _valid;
+    }
+
+protected:
+    lfs_dir_t *_getDir() const {
+        return _dir.get();
+    }
+
+    String                      _pattern;
+    LittleFSImpl               *_fs;
+    std::shared_ptr<lfs_dir_t>  _dir;
+    lfs_info                    _dirent;
+    bool                        _valid;
+    bool                        _opened;
+};
+
```

Comment:
The end of the implementation namespace should be here.

Reply:
Done latest commit